### PR TITLE
Add wrapper for motor reboot

### PR DESCRIPTION
**Pull request by zhouzypaul**
When motor fails, truncate the episode, and reboot the failed motors on reset()

### CODE REVIEW

File: manipulator_gym/utils/gym_wrappers.py [hunk 1]
```diff
@@ -5,6 +5,59 @@
 import logging
 
 
+class CheckAndRebootJoints(gym.Wrapper):
+    """
+    Every step, check whether joints have failed and reboot them if necessary.
```
Comment (youliangtan) [inline manipulator_gym/utils/gym_wrappers.py:10]: can add a short doc mention that this is now only working for widowx.

File: manipulator_gym/utils/gym_wrappers.py [hunk 2]
```diff
@@ -5,6 +5,59 @@
...
+    """
+    def __init__(self, env, interface, check_every_n_steps: int = 1):
```
Comment (youliangtan) [inline manipulator_gym/utils/gym_wrappers.py:15]: i wonder what is the overhead/latency of checking the motor every `step(action)` fn?

if it introduces significant overhead, should we have a higher default value, e.g. 10 ?

File: manipulator_gym/utils/gym_wrappers.py [hunk 3]
```diff
@@ -5,6 +5,59 @@
...
+    
+    def reset(self, **kwargs):
+        self.step_number = 0
+        
+        # reset joints on reset
+        res = self.interface.custom_fn("motor_status")
+        if res is not None:
+            for i, status in enumerate(res):
+                if status != 0:
+                    self.interface.custom_fn("reboot_motor", joint_name=self.widowx_joints[i])
```
Comment (youliangtan) [inline manipulator_gym/utils/gym_wrappers.py:56]: should we move the robot to a sleep position (or other safer position) before rebooting? is okay if it isnt too.

**Review by youliangtan — COMMENTED** [review]
Thanks for the PR

File: manipulator_gym/utils/gym_wrappers.py (hunk 2 above)
Comment (zhouzypaul) [reply manipulator_gym/utils/gym_wrappers.py:15]: In my runs I think they were pretty fast. I was just eyeballing it though (so on a precision of 1/0.1 seconds?), did not actually time it

File: manipulator_gym/utils/gym_wrappers.py (hunk 3 above)
Comment (zhouzypaul) [reply manipulator_gym/utils/gym_wrappers.py:56]: Good idea.

File: manipulator_gym/utils/gym_wrappers.py (hunk 2 above)
Comment (youliangtan) [reply manipulator_gym/utils/gym_wrappers.py:15]: nice, good to know that it is fast enough =)

**Review by youliangtan — APPROVED** [review]

**Pull request merged**